enhancements/autoscaling/provisioning-request-enablement.md

```diff
@@ -0,0 +1,337 @@
+
+Stories:
+* As an application admnistrator, I would like the Cluster Autoscaler to respect ProvisioningRequests
+  so that Kueue workloads can have infrastructure resources available when the workload is scheduled.
+  Having the ProvisioningRequest type available and the Cluster Autoscaler configured to reconcile
+  that resource will solve my problem.
+* As a cluster admnistrator who is writing automation and integrating new applications,
+  I would like to write and use tools that will create more cluster resources based on predictive resource
+  requests from my users. Being able to use ProvisioningRequests with the Cluster Autoscaler will
+  allow me to achieve my goals.
+
+### Goals
+
+* Include the `provisioningrequests.autoscaling.x-k8s.io` type in OpenShift.
+* Configure the Cluster Autoscaler to enable the controller for ProvisioningRequests.
+
+### Non-Goals
+
+* Rename the ProvisioningRequest API type, group, or version.
+* Add an API field to the ClusterAutoscaler resource to allow configuration of the ProvisioningRequest controller.
+
+## Proposal
+
+1. Add the ProvisioningRequest custom resource definition to the manifest payload for the Cluster Autoscaler Operator.
+2. Modify the Cluster Autoscaler Operator to enable the `--enable-provisioning-requests` flag of the Cluster Autoscaler.
+3. Update the Cluster Autoscaler RBAC to allow reading and writing of ProvisioningRequests.
+
+### Workflow Description
+
+This feature is exposing an API resource type and does not provide the user with additional
+controls for configuring how the resource is consumed. There are two general workflows
+that users will encounter:
+
+**cluster administrator** is a human user responsible for deploying cluster-wide changes
+for users of the cluster.
+
+**application administrator** is a human user responsible for deploying an application in
+a cluster.
+
+**Example workflow 1**
+
+1. The cluster administrator configures the Cluster Autoscaler by creating a ClusterAutoscaler
+  resource and one or more MachineAutoscaler resources.
+2. The application administrator creates a manifest for a ProvisioningRequest resource in the
+  `openshift-machine-api` namespace and applies it with the `oc` command line tool.
+3. The Cluster Autoscaler reconciles the ProvisioningRequest and begins creating the
+  requested cluster capacity.
+
+**Example workflow 2**
+
+1. The cluster administrator configures the Cluster Autoscaler by creating a ClusterAutoscaler
+  resource and one or more MachineAutoscaler resources.
+2. The cluster administrator installs and configures Kueue for cluster users, with support
+  for ProvisioningRequest enabled.
+3. The application administrator submits their workload to Kueue for scheduling.
+4. At the appropriate time, Kueue creates a ProvisioningRequest resource in the `openshift-machine-api`
+  namespace to represent the workload and the Cluster Autoscaler reconiles the resource
+  and begins to create the requested cluster capacity.
```

Comment:
I had assumed ProvReqs could be created in any namespace, is there a specific reason they must be created in this namespace? Is it possible to configure Kueue in this way? CC @kannon92

Reply:
Kueue will create a ProvReq in the namespace of the workload.

this is not configurable.

But TBH for kueue we are considering ProvReq out of scope for now. I believe this OEP was also not considering Kueue use case at the moment.

We know we need to make them work together but neither of us have the cycles to do that while both OEP are in flight..

I think there is value in proceeding with this OEP without Kueue in mind. We need these APIs enabled in Openshift to even start investigating how this will work with Kueue.

Reply:
I guess my question is, why did this namespace restriction come in in the first place, is there a way to avoid it without breaking existing autoscaler functionality

Reply:
the namespace constraint is something i introduced here as i was trying to keep towards a limited exposure for these resources like we do with other resources that can affect infrastructure (eg MachineSet, Machine, MachineAutoscaler, etc).

it appears that cluster autoscaler and kueue both have the expectation that ProvisioningRequests will be created in any namespace. i will update the enhancement.

Reply:
i updated the namespace stuff to say all namespaces